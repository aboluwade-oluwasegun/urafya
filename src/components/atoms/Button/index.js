import React, { Component } from "react";
import PropTypes from "prop-types";

class Button extends Component {
  static propTypes = {
    onClick: PropTypes.func.isRequired,
    title: PropTypes.string.isRequired,
    icon: PropTypes.string,
    showIcon: PropTypes.bool,
  };

  static defaultProps = {
    showIcon: false,
  };

  render() {
    const { icon, showIcon, onClick, title } = this.props;
    return (
      <div onClick={onClick} className="mybtncurve btn btn-primary shadow-sm">
        {showIcon && <i className={icon}></i>}
        {title}
      </div>
    );
  }
}

export default Button;
