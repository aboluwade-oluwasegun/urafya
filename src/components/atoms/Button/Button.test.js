import React from "react";
import { shallow } from "enzyme";
import Button from ".";

describe("Button component", () => {
  const shallowRender = (props = {}) =>
    shallow(
      <Button
        onClick={jest.fn()}
        title="New User"
        icon="fa fa-user"
        showIcon={true}
        {...props}
      />
    );
  it("renders correctly", () => {
    expect(shallowRender()).toMatchSnapshot();
  });

  it("renders the icon when showIcon is true", () => {
    const wrapper = shallowRender();
    expect(wrapper.find("i").exists()).toBe(true);
  });

  it("does not render the icon when showIcon is false", () => {
    const wrapper = shallowRender({ showIcon: false });
    expect(wrapper.find("i").exists()).toBe(false);
  });
  it("renders the div component", () => {
    const wrapper = shallowRender();
    expect(wrapper.find("div").exists()).toBe(true);
  });

  //test onclick later
  it("calls a function when clicked ", () => {
    const mock = jest.fn();
    const wrapper = shallowRender({ onClick: mock });
    wrapper.simulate("click");
    expect(mock).toHaveBeenCalledTimes(1);
  });
});
