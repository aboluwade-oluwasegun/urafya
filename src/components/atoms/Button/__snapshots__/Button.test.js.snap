// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Button
    icon="fa fa-user"
    onClick={[Function]}
    showIcon={true}
    title="New User"
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <i
          className="fa fa-user"
/>,
        "New User",
      ],
      "className": "mybtncurve btn btn-primary shadow-sm",
      "onClick": [Function],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "fa fa-user",
        },
        "ref": null,
        "rendered": null,
        "type": "i",
      },
      "New User",
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <i
            className="fa fa-user"
/>,
          "New User",
        ],
        "className": "mybtncurve btn btn-primary shadow-sm",
        "onClick": [Function],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "fa fa-user",
          },
          "ref": null,
          "rendered": null,
          "type": "i",
        },
        "New User",
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
  Symbol(enzyme.__childContext__): null,
}
`;
